clients/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaRequest.java

```diff
@@ -0,0 +1,431 @@
+        response = client.send(request.build(), BodyHandlers.ofInputStream());
+      } catch (HttpConnectTimeoutException e) {
+        throw new HttpClientException(
+            String.format(
+                "Timeout connecting to '%s' after %ds",
+                uri, config.getConnectionTimeoutMillis() / 1000),
+            e);
+      } catch (HttpTimeoutException e) {
+        throw new HttpClientReadTimeoutException(
+            String.format(
+                "Cannot finish %s request against '%s'. Timeout while waiting for response with a timeout of %ds",
+                method, uri, config.getReadTimeoutMillis() / 1000),
+            e);
+      } catch (MalformedURLException e) {
+        throw new HttpClientException(
+            String.format("Cannot perform %s request. Malformed Url for %s", method, uri), e);
+      } catch (IOException e) {
+        throw new HttpClientException(
+            String.format("Failed to execute %s request against '%s'.", method, uri), e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+
+      JavaResponseContext responseContext = new JavaResponseContext(response);
+
+      List<BiConsumer<ResponseContext, Exception>> callbacks = context.getResponseCallbacks();
+      if (callbacks != null) {
+        callbacks.forEach(callback -> callback.accept(responseContext, null));
+      }
+
+      config.getResponseFilters().forEach(responseFilter -> responseFilter.filter(responseContext));
+
+      if (response.statusCode() >= 400) {
+        throw new HttpClientException(
+            String.format(
+                "%s request to %s failed with HTTP/%d", method, uri, response.statusCode()));
+      }
+
+      response = null;
+      return new org.projectnessie.client.http.HttpResponse(responseContext, config.getMapper());
+    } finally {
+      if (response != null) {
+        try {
+          LOGGER.debug(
+              "Closing unprocessed input stream for {} request to {} delegating to {} ...",
+              method,
+              uri,
+              response.body());
+          response.body().close();
+        } catch (IOException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  /**
+   * Executor used to serialize the response object to JSON.
+   *
+   * <p>Java's new {@link HttpClient} uses the {@link Flow.Publisher}/{@link Flow.Subscriber}/{@link
+   * Flow.Subscription} mechanism to write and read request and response data. We have to use that
+   * protocol. Since none of the implementations must not block, writes and reads run in a separate
+   * pool.
+   *
+   * <p>Jackson has no "reactive" serialization mechanism, which means that we have to provide a
+   * custom {@link OutputStream}, which delegates {@link Flow.Subscriber#onNext(Object) writes} to
+   * the subscribing code.
+   */
+  private static final Executor writerPool =
+      new ForkJoinPool(Math.max(8, ForkJoinPool.getCommonPoolParallelism()));
```

Comment:
Make the `8` configurable just in case?

Reply:
Better not. It's there to allow _at least_ 8 concurrent request serializations on tiny machines - that's already a lot.

Reply:
fair enough